core/services/fluxmonitor/flux_monitor.go

```diff
@@ -617,6 +614,20 @@ func (p *PollingDeviationChecker) pollIfEligible(threshold float64) (createdJobR
 		return false
 	}
 
+	available, err := p.fluxAggregator.GetAvailableFunds()
+	if err != nil {
+		logger.Errorf("unable to determine available funds from FluxAggregator contract : %v", err)
+		return false
+	}
+
+	if available.Cmp(p.store.Config.MinimumContractPayment()) < 0 {
```

Comment:
There's a `paymentAmount` field in the `RoundDetails` struct of the FluxAggregator contract -- that's what we need to be comparing against here.  Making sure that that value is `>=` the node's minimum contract payment might also be a good idea, but I would argue that that kind of validation should go in a separate story, as it might be impacted by the forthcoming service agreement work.  Maybe we should sync up with @se3000 about that.

Reply:
I found discovering which field to use for 'minimumPayment' problematic.

Reply:
I think it makes sense to check both. You need to know both that the payment amount is enough(`RoundDetails.paymentAmount`) and that the funds of the contract are enough to cover that(`GetAvailableFunds()`).